src/js/components/Tabs/Tabs.js

```diff
@@ -103,19 +234,154 @@ const Tabs = forwardRef(
+            flex={!!overflow}
+            wrap={false}
+            overflow={overflow ? 'hidden' : 'visible'}
+            background={theme.tabs.header.background}
+            gap={theme.tabs.gap}
+            pad={overflow ? '2px' : undefined}
+            margin={overflow ? '-2px' : undefined}
+          >
+            {tabs}
+          </StyledTabsHeader>
+          {overflow && (
+            <Button
+              a11yTitle="Next Tab"
+              icon={<Next />}
+              disabled={disableRightArrow}
+              // removed from tabIndex, button is redundant for keyboard users
+              tabIndex={-1}
+              onClick={() => {
+                let scrolledToIndex;
+                let i = tabRefs.length - 1;
+
+                while (scrolledToIndex === undefined && i > 0) {
+                  if (
+                    !isVisible(tabRefs[i].current) &&
+                    isVisible(tabRefs[i - 1].current)
+                  ) {
+                    const headerRect =
+                      headerRef.current.getBoundingClientRect();
+                    for (let j = 0; j < step; j += 1) {
+                      if (i <= tabRefs.length - 1) {
+                        const tabRect =
+                          tabRefs[i].current?.getBoundingClientRect();
+                        let amountHidden = 0;
+                        if (
+                          tabRect.left >= headerRect.left - 1 &&
+                          tabRect.left <= headerRect.right + 1
+                        ) {
+                          amountHidden =
+                            tabRect.width - (headerRect.right - tabRect.left);
+                        } else {
+                          amountHidden = tabRect.width;
+                        }
+                        if (j === step - 1 || j === tabRefs.length - 1)
+                          headerRef.current.scrollBy({
+                            left: amountHidden,
+                            behavior: 'smooth',
+                          });
+                        else
+                          headerRef.current.scrollBy({
+                            left: amountHidden,
+                          });
+                        scrolledToIndex = i;
+                        i += 1;
+                      }
+                    }
+                  }
+                  i -= 1;
+                }
+                setDisableLeftArrow(false);
+                if (scrolledToIndex === tabRefs.length - 1) {
+                  // wait for scroll animation to finish
+                  const checkVisible = setInterval(() => {
+                    if (isVisible(tabRefs[tabRefs.length - 1].current)) {
+                      setDisableRightArrow(true);
+                      clearInterval(checkVisible);
+                    }
+                  }, 100);
+                  setTimeout(() => {
+                    clearInterval(checkVisible);
+                  }, 500);
```

Comment:
can you add a comment about how these timeout values were determined?

Reply:
Added comments for this in latest commit